app/src/main/java/com/example/nvdovin/weatherapp/adapter/RecyclerViewAdapter.java

```diff
@@ -95,13 +95,13 @@ private int setWeatherIcon(String id, TextView iconTextView) {
         return resID;
     }
 
-    public void swap(List<ForecastNow> forecastNowList) {
+    public void swap(List<CurrentForecast> currentForecastList) {
 
-        if (this.forecastNowList != null) {
-            this.forecastNowList.clear();
-            this.forecastNowList.addAll(forecastNowList);
+        if (this.currentForecastList != null) {
```

Comment:
you can always do list=new Arraylist(sourceList) and get rid of the entire "if" block.